### PR TITLE
[FCE-429] Rename peer to participant

### PR DESCRIPTION
**Pull request by mironiasty**
## Description

Rename `usePeers` to `useParticipants` (with all related stuff on JS side).

Also update metadata for participant to be explicit map with string keys. This way user can type it in his app, but it won't be required.

## Motivation and Context

After some discussions, we decided that Participant will be better name for our users.

## How has this been tested?

I just hope it works 🙃 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/react-native-client/src/hooks/useCamera.ts [hunk 1]
```diff
@@ -77,7 +76,7 @@ export type CameraConfigInternal = CameraConfigBase & {
   /**
    * a map `string -> any` containing video track metadata to be sent to the server.
    */
-  videoTrackMetadata?: Metadata;
+  videoTrackMetadata?: { active: boolean; type: 'camera' };
```
Comment (kamil-stasiak) [inline packages/react-native-client/src/hooks/useCamera.ts:79]: This js-doc is no longer accurate

**Review by kamil-stasiak — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -279,17 +279,17 @@ await startMicrophone({ audioTrackMetadata: { active: true, type: "audio" } });
 For more options and functions to control the camera and microphone see the API
 documentation.
 
-If you have the connection set up, then use `usePeers()` hook to track the
-other peers in the room. One of the peers will be a local participant
-(the one who's using the device). When peers is added or removed because an
-user joins or leaves the room, the peers will be updated automatically.
+If you have the connection set up, then use `useParticipiants()` hook to track the
```
Comment (czerwiukk) [inline README.md:282]: **typo:** participants

**Review by czerwiukk — CHANGES_REQUESTED** [review]

File: examples/video-chat/screens/PreviewScreen/PreviewScreen.tsx [hunk 3]
```diff
@@ -101,9 +102,13 @@ function PreviewScreen({
   }, [getCaptureDevices, startCamera]);
 
   const onJoinPressed = async () => {
-    await connect(route.params.fishjamUrl, route.params.peerToken, {
-      name: route.params.userName,
-    });
+    await connect<ParticipiantMetadata>(
+      route.params.fishjamUrl,
+      route.params.peerToken,
```
Comment (karkakol) [inline examples/video-chat/screens/PreviewScreen/PreviewScreen.tsx:107]: I think `route.params.peerToken` be renamed to `route.params.participant`

File: examples/video-chat/screens/RoomScreen.tsx [hunk 4]
```diff
@@ -46,9 +47,12 @@ const RoomScreen = ({ navigation, route }: Props) => {
 
   useForegroundService();
 
-  const { peers } = usePeers();
+  const { participiants } = useParticipiants<ParticipiantMetadata>();
```
Comment (karkakol) [inline examples/video-chat/screens/RoomScreen.tsx:50]: typo:
`participiants` -> `participants`

File: examples/webdriverio-test/test/specs/test.e2e.ts [hunk 5]
```diff
@@ -68,47 +68,50 @@ const createFishjamRoom = async () => {
...
   }
 };
-const addPeerToRoom = async (
+const addParticipiantToRoom = async (
   roomId: string,
   enableSimulcast: boolean = true,
 ) => {
-  const { addPeer } = RoomApiFp(config);
-  const addPeerRequest: AddPeerRequest = {
+  const { addPeer: addParticipiant } = RoomApiFp(config);
```
Comment (karkakol) [inline examples/webdriverio-test/test/specs/test.e2e.ts:75]: Do we want to rename `@fishjam-cloud/fishjam-openapi` functions to `Participants`?

**Review by karkakol — APPROVED** [review]

File: examples/video-chat/components/DismissKeyboard.tsx [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { TouchableWithoutFeedback, Keyboard } from 'react-native';
 
-import type { AppParentNode } from '../types/AppParentNode';
+import type { AppParentNode } from '../types/types';
```
[the remark this answers is not in the document]
Comment (mironiasty) [reply examples/video-chat/components/DismissKeyboard.tsx:4]: I remove `AppParentNode` type and renamed file to `metadata.ts`. At some day, we might do another change in types structure here 🤔

File: examples/video-chat/screens/PreviewScreen/PreviewScreen.tsx (hunk 3 above)
Comment (mironiasty) [reply examples/video-chat/screens/PreviewScreen/PreviewScreen.tsx:107]: That is interesting problem: how we should call peer token now?

File: examples/video-chat/screens/RoomScreen.tsx (hunk 4 above)
Comment (mironiasty) [reply examples/video-chat/screens/RoomScreen.tsx:50]: 🙈

File: examples/webdriverio-test/test/specs/test.e2e.ts (hunk 5 above)
Comment (mironiasty) [reply examples/webdriverio-test/test/specs/test.e2e.ts:75]: maybe someday?  there are few other tickets for renames

**Review by czerwiukk — APPROVED** [review]

**Review by karkakol — APPROVED** [review]
👍

**Pull request merged**